src/commands/connections/deleteCosmosDBConnection.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionAccountDatabaseTreeItem } from 'src/explorer/ConnectionAccountDatabaseTreeItem';
+import * as vscode from 'vscode';
+import { IAzureNode } from 'vscode-azureextensionui';
+import * as constants from '../../constants';
+import { IConnections } from './IConnections';
+
+export async function deleteCosmosDBConnection(node: IAzureNode<ConnectionAccountDatabaseTreeItem>): Promise<void> {
+    const connectionToDelete = node.treeItem.connectionId;
+    const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
+    const allConnections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
+
+    let connectionsUnit = allConnections.find((x: IConnections) => x.webAppId === node.treeItem.client.id);
+    // tslint:disable-next-line:strict-boolean-expressions
+    connectionsUnit = connectionsUnit || <IConnections>{};
```

Comment:
If connectionsUnit isn't defined, then you don't need to remove anything. Same for connectionsUnit.cosmosDB

I would just wrap the rest of this code in `if (connectionsUnit && connectionsUnit.cosmosDB) {`